module/zfs/dsl_scan.c

```diff
@@ -4287,35 +4290,49 @@ dsl_scan_sync(dsl_pool_t *dp, dmu_tx_t *tx)
 	dsl_scan_t *scn = dp->dp_scan;
 	spa_t *spa = dp->dp_spa;
 	state_sync_type_t sync_type = SYNC_OPTIONAL;
+	uint64_t to_issue, issued;
+	int restart_early;
 
 	if (spa->spa_resilver_deferred &&
 	    !spa_feature_is_active(dp->dp_spa, SPA_FEATURE_RESILVER_DEFER))
 		spa_feature_incr(spa, SPA_FEATURE_RESILVER_DEFER, tx);
 
+	/*
+	 * Only process scans in sync pass 1.
+	 */
+	if (spa_sync_pass(spa) > 1)
+		return;
+
+	/*
+	 * issued/to_issue as presented to the user
+	 * in print_scan_scrub_resilver_status() issued/total_i
+	 * @ cmd/zpool/zpool_main.c
+	 */
+	to_issue = scn->scn_phys.scn_to_examine - scn->scn_phys.scn_skipped;
+	issued = scn->scn_issued_before_pass + spa->spa_scan_pass_issued;
+	restart_early = spa->spa_resilver_deferred && (
+	    zfs_resilver_disable_defer ||
+	    (issued < (to_issue * zfs_resilver_defer_percent / 100)));
```

Comment:
I'd suggest initializing `restart_early` above and then reworking this check to pull it under a `spa->spa_resilver_deferred` conditional.  Something like this:

```c
        if (spa->spa_resilver_deferred) {
                uint64_t to_issue = scn->scn_phys.scn_to_examine - 
                    scn->scn_phys.scn_skipped;
                uint64_t issued = scn->scn_issued_before_pass + 
                    spa->spa_scan_pass_issued;

                restart_early = zfs_resilver_disable_defer ||
                    (issued < (to_issue * zfs_resilver_defer_percent / 100));
        }
```

You can drop the comment too since `restart_early` is explained in the block comment below.

Reply:
ok but now I'm unsure whether I got the formatting right, it spans behind the 80 cols mark, so I split the lines a bit awkwardly IMO... let's hope next force-push won't make these comments disappear